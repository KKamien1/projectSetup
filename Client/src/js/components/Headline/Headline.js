import React, { useState, useEffect } from 'react';
import './Headline.scss';

import { setTimeout } from 'timers';

export default function Headline({ children, delay = 60 }) {
  const [classes, setClasses] = useState('letter');

  useEffect(() => {
    setTimeout(() => {
      setClasses('letter letter--animated');
    }, 150);
  }, [classes]);

  const html = Array.from(children).map((letter, index) => {
    const minWidth = letter === ' ' ? '.5em' : 'auto';
    return (
      <span
        key={letter + index}
        className={classes}
        style={{ minWidth, transitionDelay: `${index * delay}ms` }}
      >
        {letter}
      </span>
    );
  });

  return <div className="headline">{html}</div>;
}
