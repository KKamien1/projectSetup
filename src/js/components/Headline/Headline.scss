@import "../../../sass/variables.sass";

.headline {
  font-size: 2rem;
  color: inherit;
}

.letter {
  color: $muted;
  display: inline-block;
  transition: 0.2s;
  opacity: 0.5;
  transform: translateY(20px);
}

.letter--animated {
  color: inherit;
  opacity: 0.9;
  transform: translateY(0px);
}
