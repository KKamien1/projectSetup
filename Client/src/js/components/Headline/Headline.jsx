import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import './Headline.scss';

import { setTimeout } from 'timers';

export default function Headline({ children, delay = 60 }) {
  const [classes, setClasses] = useState('letter');

  useEffect(() => {
    setTimeout(() => {
      setClasses('letter letter--animated');
    }, 150);
  }, [classes]);

  const html = Array.from(children).map((letter, index) => {
    const minWidth = letter === ' ' ? '.5em' : 'auto';
    const key = `${letter + index}`;
    return (
      <span
        key={key}
        className={classes}
        style={{ minWidth, transitionDelay: `${index * delay}ms` }}
      >
        {letter}
      </span>
    );
  });

  return <div className="headline">{html}</div>;
}

Headline.propTypes = {
  children: PropTypes.node.isRequired,
  delay: PropTypes.number,
};
Headline.defaultProps = {
  delay: 60,
};
