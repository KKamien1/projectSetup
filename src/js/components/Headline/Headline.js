import React, { useState, useEffect, useRef } from "react";
import "./Headline.scss";

import useDebounce from "../../utils/useDebounce";
import { setTimeout } from "timers";

export default function Headline(props) {
  const { children, delay = 60 } = props;

  const [classes, setClasses] = useState("letter");

  //const debouncedAddClass = useDebounce(classes, 1000);

  useEffect(() => {
    setTimeout(() => {
      setClasses("letter letter--animated");
    }, 1000);
  }, [classes]);

  const html = Array.from(children).map((letter, index) => {
    let minWidth = letter === " " ? ".5em" : "auto";
    return (
      <span
        key={letter + index}
        className={classes}
        style={{ minWidth, transitionDelay: `${index * delay}ms` }}
      >
        {letter}
      </span>
    );
  });

  return <div className="headline">{html}</div>;
}
