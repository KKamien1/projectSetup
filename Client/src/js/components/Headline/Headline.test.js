import React from 'react';
import { mount, shallow } from 'enzyme';
import { mountWithIntl, shallowWithIntl } from "enzyme-react-intl";
import Headline from "./Headline.jsx";

describe('Headline component testing', () => {
  let wrapper;

  beforeEach(() => {
    store = { menu: [] };
    wrapper = shallow(<Headline />).dive();
  });

  it('Metches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
  });

});
