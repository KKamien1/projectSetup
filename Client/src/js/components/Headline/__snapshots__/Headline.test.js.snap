// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Headline component testing Metches the snapshot 1`] = `undefined`;
